modules/blockbuilder/blockbuilder.go

```diff
@@ -280,8 +291,13 @@ func (b *BlockBuilder) consume(ctx context.Context) (time.Duration, error) {
 }
 
 func (b *BlockBuilder) consumePartition(ctx context.Context, ps partitionState) (lastTs time.Time, commitOffset int64, err error) {
+	ctx, span := tracer.Start(ctx, "blockbuilder.consumePartition",
```

Comment:
We are detaching the link between the distributor and the blockbuilder.
I was wondering if we could take the context from the first record in the consumption loop and create a span from it:

https://github.com/twmb/franz-go/blob/master/plugin/kotel/README.md#processing-records

That way we could track the whole journey

Reply:
I've added a link to the producer span, so block-builder traces are linked to producing traces.

Producing traces and block-builder traces is a many-to-many problem, if we were to build them as a single trace I'm not sure how we would build them without them being chaotic.